shared/js/loadApp.js

```diff
@@ -23,3 +84,9 @@ module.exports = function (callback) {
     callback();
   }
 };
+
+function setLastAttemptUnlessJigsaw(source) {
+  if (appOptions.levelGameName !== 'Jigsaw') {
+    appOptions.level.lastAttempt = source;
+  }
+}
```

Comment:
Being conservative here, the correct place for this code will be `StudioApp.js` after HoC (see: https://github.com/code-dot-org/code-dot-org/pull/5740).